### PR TITLE
Update ERC-4824: Move to Review

### PR DESCRIPTION
**Pull request by thelastjosh**
I moved the example contract to the assets folder, made various minor clarifications about how daoURI and indexing should work, supplemented the rationale, changed the id format for members to align more closely with CAIP-10 (already used for proposals), and fixed a bunch of typos. All changes are backward compatible.

### CODE REVIEW

File: ERCS/erc-4824.md [hunk 1]
```diff
@@ -53,9 +53,7 @@ The DAO JSON-LD Schema mentioned above:
 }
 ```
 
-A DAO MAY inherit the above interface above or it MAY create an external registration contract that is compliant with this EIP. If a DAO creates an external registration contract, the registration contract MUST store the DAO’s primary address. 
-
-If the DAO inherits the above interface, it SHOULD define a method for updating daoURI. If the DAO uses an external registration contract, the registration contract SHOULD contain some access control logic to enable efficient updating for daoURI.
+A DAO MAY inherit the interface above or it MAY create an external registration contract that is compliant with this EIP. If a DAO creates an external registration contract, the registration contract MUST store the DAO’s primary address.
 
 ```solidity
```
Comment (SamWilsn) [inline ERCS/erc-4824.md:58]: I _think_ this contract should be in the reference implementation section. It doesn't sound like the internals of this contract need to be exactly this, and any compatible implementation would work just as well. I'd recommend making this an interface, and moving the implementation to the reference implementation section.

---

Also your contract name seems to be a victim of find/replace.

File: ERCS/erc-4824.md [hunk 2]
```diff
@@ -396,17 +330,21 @@ DAOs themselves have a number of existing and emerging use-cases. But almost all
...
 
 **Proposal ID**. Every unique on-chain proposal MUST be associated to a proposal ID of the form CAIP10_ADDRESS + “?proposalId=” + PROPOSAL_COUNTER, where PROPOSAL_COUNTER is an arbitrary string which is unique per CAIP10_ADDRESS. Note that PROPOSAL_COUNTER may not be the same as the on-chain representation of the proposal; however, each PROPOSAL_COUNTER should be unique per CAIP10_ADDRESS, such that the proposal ID is a globally unique identifier. We endorse the CAIP-10 standard to support multi-chain / layer 2 proposals and the “?proposalId=” query syntax to suggest off-chain usage.
```
Comment (SamWilsn) [inline ERCS/erc-4824.md:349]: Be careful to keep requirements (usually indicated with the all-caps keywords) in the specification section.

File: assets/erc-4824/contracts/ERC4824Factory.sol [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            mstore(
+                add(clone, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            result := create(0, clone, 0x37)
+        }
+    }
+}
+
+contract ERC-4824RegistrationSummoner {
```
Comment (SamWilsn) [inline assets/erc-4824/contracts/ERC4824Factory.sol:26]:
```suggestion
contract ERC4824RegistrationSummoner {
```

Maybe?

File: assets/erc-4824/contracts/ERC4824Factory.sol [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    event NewRegistration(
+        address indexed daoAddress,
+        string daoURI,
+        address registration
+    );
+
+    address public ERC-4824Index;
```
Comment (SamWilsn) [inline assets/erc-4824/contracts/ERC4824Factory.sol:33]:
```suggestion
    address public ERC4824Index;
```

File: ERCS/erc-4824.md (hunk 1 above)
Comment (thelastjosh) [reply ERCS/erc-4824.md:58]: moved the contract code to assets in the new commit, and fixed the find/replace error with the double title

File: ERCS/erc-4824.md [hunk 5]
```diff
@@ -356,36 +258,68 @@ Activity Log JSON-LD Schema. Every contract implementing this EIP SHOULD impleme
 
 Contracts JSON-LD Schema. Every contract implementing this EIP SHOULD implement a contractsURI pointing to a JSON object satisfying this schema.
 
-Further, every contractsURI SHOULD include at least the contract inheriting the ERC-4824 interface.
+`contractsURI` is especially important for DAOs with distinct or decentralized governance occurring across multiple different contracts, possibly across several chains. Multiple addresses may report the same daoURI.
```
[the remark this answers is not in the document]
Comment (thelastjosh) [reply ERCS/erc-4824.md:261]: I've removed the backticks except in cases where it was necessary for clarity in the next (e.g. comparing a `members` field with a `membersURI` field).

File: ERCS/erc-4824.md (hunk 2 above)
Comment (thelastjosh) [reply ERCS/erc-4824.md:349]: fixed!

**Comment by thelastjosh** [comment]
Responded to all review suggestions, moved the smart contract out into assets. Should be ready to move to review!

**Comment by thelastjosh** [comment]
@SamWilsn @xinbenlv or other EIP editor, just a ping so this change can get finally merged! 🙏

**Comment by thelastjosh** [comment]
> Put up a pull request with some minor fixes: [thelastjosh#3](https://github.com/thelastjosh/ERCs/pull/3)

Merged!

**Review by SamWilsn — APPROVED** [review]

**Comment by thelastjosh** [comment]
Thank you!!

**Pull request merged**